### PR TITLE
change the script for the mutation classifications in Destabilizing n…

### PR DESCRIPTION
**Pull request by marnaudi**
…eutral or Stabilizing.

### CODE REVIEW

**Comment by marnaudi** [comment]
The paths in which the pipeline takes the input files are different because I tested in my folder. When everything is checked I will change the path with the original ones

File: SNOmodels/Snakefile [hunk 1]
```diff
@@ -24,9 +26,43 @@ from MDAnalysis.analysis.dihedrals import Dihedral
...
+#as input a value of pKa of a SNO site upon mutation and a list of parameter from the WT condition containing the pKa value of the SNO site as long as the distance
+#between the SNO cysteine and the proximal cysteine in the WT and mutant form.
+#substracts the WT SNO site pKa value from the mutant's one.
+#substracts the WT distance between the two cysteines from the distance calculated for the mutants
+#return Stabilizing if the calculated difference is lower than -1, Destabilizing if the differences is higher than 1, Neutral if none of the previous cases happens.
+
+def assignment_based_on_distance_and_pKa(d_SNO_sg_CYS_sg_avg,pKa_SNO_avg,par_WT_list):
+    distance_diff=d_SNO_sg_CYS_sg_avg-par_WT_list[0]
+    pKa_diff=pKa_SNO_avg-par_WT_list[1]
+    if distance_diff>=1 and pKa_diff>=1:
```
Comment (mtiberti) [inline SNOmodels/Snakefile:56]: these cut-offs should be user-changeable parameters to set at the beginning of the Snakefile; same for other function

File: SNOmodels/Snakefile [hunk 2]
```diff
@@ -368,6 +405,34 @@ rule make_statistics:
...
+        for i in gene_list:
+            df_i=df.loc[df["hugo"]==wildcards.hugo]
+            proxi_list=df_i.loc[df_i["mutation"]=="WT","proximal_Cys"].to_list()
+            output_df=pd.DataFrame(columns=columns)
+            for prox in proxi_list:
+                df_prox_cys=df_i.loc[df_i["proximal_Cys"]==prox]
+                subset_df_prox_cys=df_prox_cys[["d_SNO-sg_CYS-sg_avg","pKa_SNO_avg","mutation","proximal_Cys"]]
+                par_WT_list=subset_df_prox_cys.loc[(subset_df_prox_cys["proximal_Cys"]==prox)&(subset_df_prox_cys["mutation"]=="WT")].values.flatten().tolist()
+                df_prox_cys['pKa_classification'] = df_i.apply(lambda x: assignment_based_on_pKa(x["pKa_SNO_avg"],par_WT_list), axis=1)
+                df_prox_cys['distance_pKa_classification'] = df_i.apply(lambda x: assignment_based_on_distance_and_pKa(x["d_SNO-sg_CYS-sg_avg"],x["pKa_SNO_avg"],par_WT_list), axis=1)
```
Comment (mtiberti) [inline SNOmodels/Snakefile:428]: what happens with WT in this case? In your output you also classify the WT cases which shouldn't be classified

**Review by mtiberti — CHANGES_REQUESTED** [review]

File: SNOmodels/Snakefile [hunk 3]
```diff
@@ -1,19 +1,21 @@
-# SNOmodels - pipeline for analyzing cysteine residues from structural ensembles
-# Copyright (C) 2022 Matteo Tiberti, Danish Cancer Society
+#SNOmodels - pipeline for analyzing cysteine residues from structural ensembles
+# Copyright (C) 2022 Matteo Tiberti, Danish Cancer Society
```
[the remark this answers is not in the document]
Comment (marnaudi) [reply SNOmodels/Snakefile:2]: don

File: SNOmodels/Snakefile [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
 # This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+
```
[the remark this answers is not in the document]
Comment (marnaudi) [reply SNOmodels/Snakefile:17]: done

File: SNOmodels/Snakefile [hunk 5]
```diff
@@ -24,9 +26,43 @@ from MDAnalysis.analysis.dihedrals import Dihedral
 from collections import defaultdict
 import re
 
-cys_residues_csv = '../SNOfinder/results/cys_database_far_seq_acc10_human.csv'
+#functions for the variants classificstion based on pKa changes of the SNO site and distance changes between SNO and proxymal cysteine upon mutation:
+
+
+#assignment_based_on_pKa function takes:
```
[the remark this answers is not in the document]
Comment (marnaudi) [reply SNOmodels/Snakefile:32]: done

File: SNOmodels/Snakefile [hunk 6]
```diff
@@ -24,9 +26,43 @@ from MDAnalysis.analysis.dihedrals import Dihedral
...
+def assignment_based_on_pKa(pKa_SNO_avg,par_WT_list):
+    pKa_diff=pKa_SNO_avg-par_WT_list[1]
+    if pKa_diff>=1:
+        return "Destabilizing"
+    if pKa_diff<=-1:
+        return "Stabilizing"
+    else:
+        return "Neutral"
+
+#assignment_based_on_distance_and_pKa function takes:
```
[the remark this answers is not in the document]
Comment (marnaudi) [reply SNOmodels/Snakefile:46]: done

File: SNOmodels/Snakefile (hunk 1 above)
Comment (marnaudi) [reply SNOmodels/Snakefile:56]: done

File: SNOmodels/Snakefile [hunk 7]
```diff
@@ -368,6 +405,34 @@ rule make_statistics:
...
+            df_i=df.loc[df["hugo"]==wildcards.hugo]
+            proxi_list=df_i.loc[df_i["mutation"]=="WT","proximal_Cys"].to_list()
+            output_df=pd.DataFrame(columns=columns)
+            for prox in proxi_list:
+                df_prox_cys=df_i.loc[df_i["proximal_Cys"]==prox]
+                subset_df_prox_cys=df_prox_cys[["d_SNO-sg_CYS-sg_avg","pKa_SNO_avg","mutation","proximal_Cys"]]
+                par_WT_list=subset_df_prox_cys.loc[(subset_df_prox_cys["proximal_Cys"]==prox)&(subset_df_prox_cys["mutation"]=="WT")].values.flatten().tolist()
+                df_prox_cys['pKa_classification'] = df_i.apply(lambda x: assignment_based_on_pKa(x["pKa_SNO_avg"],par_WT_list), axis=1)
+                df_prox_cys['distance_pKa_classification'] = df_i.apply(lambda x: assignment_based_on_distance_and_pKa(x["d_SNO-sg_CYS-sg_avg"],x["pKa_SNO_avg"],par_WT_list), axis=1)
+                output_df=pd.concat([output_df,df_prox_cys])
```
[the remark this answers is not in the document]
Comment (marnaudi) [reply SNOmodels/Snakefile:429]: removed

File: SNOmodels/Snakefile [hunk 8]
```diff
@@ -368,6 +405,34 @@ rule make_statistics:
 
         pd.concat(out_df, axis='rows').to_csv(output.stats_csv, index=None, float_format='%.3f')
 
+rule mutation_classification:
```
[the remark this answers is not in the document]
Comment (marnaudi) [reply SNOmodels/Snakefile:408]: done

File: SNOmodels/Snakefile (hunk 2 above)
Comment (marnaudi) [reply SNOmodels/Snakefile:428]: done

**Pull request merged**